src/components/buttons/browse-the-book.cjsx

```diff
@@ -13,23 +14,48 @@ module.exports = React.createClass
 
   propTypes:
     courseId:  React.PropTypes.string
+    ecosystemId:  React.PropTypes.string
     chapterId: React.PropTypes.number
     sectionId: React.PropTypes.number
     page:      React.PropTypes.string
     unstyled:  React.PropTypes.bool
     bsStyle:   React.PropTypes.string
 
+  getLinkProps: (link) ->
+    linkProps = {target:'_blank', className:'view-reference-guide', href: link}
+    linkProps.className += " #{@props.className}" if @props.className
+    text = @props.children or 'Browse the Book'
+
+    omitProps = _.chain(@propTypes)
+      .keys()
+      .union(['children', 'className', 'unstyled'])
+      .value()
+    # sometimes, props should transfer, such as onClick
+    transferProps = _.omit(@props, omitProps)
+
+    _.extend({}, transferProps, linkProps)
+
   render: ->
     courseId = @props.courseId or @context.router.getCurrentParams().courseId
 
     return null unless courseId # if we don't have a course id we can't browse it's book
 
+    unless @props.page
+      {ecosystemId} = @props
+      courseEcosystemId = CourseStore.get(courseId)?.ecosystem_id
+
+      # only link with ecosystem id if needed.
+      if ecosystemId isnt courseEcosystemId
+        queryParams = {ecosystemId}
+
     # the router is smart enough to figure out which props are present and return the best route
     linkType = if @props.page then 'viewReferenceBookPage' else
       if @props.section then 'viewReferenceBookSection' else 'viewReferenceBook'
-    link = @context.router.makeHref( linkType, {courseId, cnxId: @props.page, section:@props.section} )
-    linkProps = {target:'_blank', className:'view-reference-guide', href: link}
+    link = @context.router.makeHref( linkType, {courseId, cnxId: @props.page, section:@props.section}, queryParams )
```

Comment:
I think it's because the `link` variable is used to build up the `linkProps` which is then used in either a plain `<a />` tag, or a `BS.Button` depending on whether the `unstyled` prop is set.   The plain anchor tag could be switched out for a `Router.Link`, and the BS.Button could be as well if we added the bootstrap styles to it.  

I also see that the specs check that `context.router.makeHref` is called with appropriate arguments, but I think they could also be swapped to check Router.Link.